onadata/apps/logger/models/xform.py

```diff
@@ -953,8 +953,17 @@ def soft_delete(self, user=None):
             update_fields.append('deleted_by')
 
         self.save(update_fields=update_fields)
+        # Delete associated filtered datasets
         for dataview in self.dataview_set.all():
             dataview.soft_delete(user)
```

Comment:
I wonder if we have a class function that can be used to un-delete an XForm? Seems like a nice to have IMO since this complicates the recovery process a bit

Reply:
Yeah, so these changes do increase complexity if a form is to be restored. I agree that we may need a management command that can reverse this, maybe a different issue can capture this?
To uncomplicate this though, I felt the inline comments help one understand the difference of linked datasets and merged datasets.
I also only removed the form media files, I left the other form metadata e.g enketo links, in the event a user may want to have the form restored.

Reply:
Yea, we don't have to implement a migration command for this... I think an instance method should do the trick. I'm fine with this being included in a separate PR.

I think re-creating the enketo links is something we can easily do on restoration of the form... The media files I'm iffy on